pkg/controller/wip/controller.go

```diff
@@ -119,8 +125,151 @@ func (c *controller) brokerDelete(obj interface{}) {
+			c.updateBrokerReadyCondition(broker, v1alpha1.ConditionFalse, ErrorSyncingCatalogReason, ErrorSyncingCatalogMessage)
+			return
+		} else {
+			glog.V(5).Infof("Reconciled serviceClass %v (broker %v)", serviceClass.Name, broker.Name)
+		}
+	}
+
+	err = c.updateBrokerReadyCondition(broker, v1alpha1.ConditionTrue, "FetchedCatalog", "Successfully fetched catalog from broker")
+	if err != nil {
+		glog.Errorf("Error updating ready condition for Broker %v: %v", broker.Name, err)
+	} else {
+		glog.V(5).Infof("Updated ready condition for Broker %v to True", broker.Name)
+	}
+}
+
+// reconcileServiceClassFromBrokerCatalog reconciles a ServiceClass after the
+// Broker's catalog has been re-listed.
+func (c *controller) reconcileServiceClassFromBrokerCatalog(broker *v1alpha1.Broker, serviceClass *v1alpha1.ServiceClass) error {
+	serviceClass.BrokerName = broker.Name
+
+	existingServiceClass, err := c.serviceClassLister.Get(serviceClass.Name)
+	if err != nil {
+		// An error returned from a lister Get call means that the object does
+		// not exist.  Create a new ServiceClass.
+		if _, err := c.serviceCatalogClient.ServiceClasses().Create(serviceClass); err != nil {
+			glog.Errorf("Error creating serviceClass %v from Broker %v: %v", serviceClass.Name, broker.Name, err)
+			return err
+		}
+	}
+
+	clone, err := api.Scheme.DeepCopy(existingServiceClass)
+	if err != nil {
+		return err
+	}
+
+	toUpdate := clone.(*v1alpha1.ServiceClass)
+	toUpdate.Bindable = serviceClass.Bindable
+	toUpdate.Plans = serviceClass.Plans
+	toUpdate.PlanUpdatable = serviceClass.PlanUpdatable
+	toUpdate.OSBTags = serviceClass.OSBTags
+	toUpdate.OSBRequires = serviceClass.OSBRequires
+	toUpdate.OSBMaxDBPerNode = serviceClass.OSBMaxDBPerNode
+	toUpdate.OSBDashboardOAuth2ClientID = serviceClass.OSBDashboardOAuth2ClientID
+	toUpdate.OSBDashboardSecret = serviceClass.OSBDashboardSecret
+	toUpdate.OSBDashboardRedirectURI = serviceClass.OSBDashboardRedirectURI
+
+	toUpdate.Description = serviceClass.Description
+	toUpdate.DisplayName = serviceClass.DisplayName
+	toUpdate.ImageURL = serviceClass.ImageURL
+	toUpdate.LongDescription = serviceClass.LongDescription
+	toUpdate.ProviderDisplayName = serviceClass.ProviderDisplayName
+	toUpdate.DocumentationURL = serviceClass.DocumentationURL
+	toUpdate.SupportURL = serviceClass.SupportURL
+
+	if _, err := c.serviceCatalogClient.ServiceClasses().Update(toUpdate); err != nil {
+		glog.Errorf("Error updating serviceClass %v from Broker %v: %v", serviceClass.Name, broker.Name, err)
+		return err
+	}
+
+	return nil
+}
+
+func (c *controller) updateBrokerReadyCondition(broker *v1alpha1.Broker, status v1alpha1.ConditionStatus, reason, message string) error {
+	var (
+		foundReadyCondition = false
+		readyConditionIndex = 0
+		newCondition        = v1alpha1.BrokerCondition{
+			Type:    v1alpha1.BrokerConditionReady,
+			Status:  status,
+			Reason:  reason,
+			Message: message,
+		}
+	)
+
+	for ii, condition := range broker.Status.Conditions {
```

Comment:
Why `ii` instead of `i`? Looks odd... I'm distracted wondering what it might stand for and searching above to see if `i` is already used elsewhere. Good code limits such distractions.